### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by patrick-hovsepian**
@ConradSelig This builds off of my previous PR and I am just seeing your comments there. 

This PR is mainly for performance improvements - the biggest change is fetching all ticker data in one request rather than a request per ticker.

If you're not opposed, I'll work the feedback from the previous PR into this one and just close that.

I'm aware of the merge conflicts and will address those - in the meantime, wanted to get your eyes on this.

### CODE REVIEW

File: sample-portfolio.ini
Comment (patrick-hovsepian) [inline sample-portfolio.ini]: I don't think we need a gen sample config function. I've removed it and replaced it with a static sample file

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:189]: make this a map for faster lookups

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: single download for all ticker data. will wrap this in a try catch per the other PR that was just merged

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:329]: not entirely sure what this was doing - was it akin to https://github.com/ConradSelig/cliStocksTracker/pull/30/files?file-filters%5B%5D=.py#diff-6d3d1c3e08993dd25420da259e7dc3b20dc52ae0b2dda25ba2644fe6a2c7d471R600

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: oops, will remove this

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: same as previous PR

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: added the current timestamp when printing

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: see the comment above but this accounts for potential NaN values in ticker data. There should be a more efficient way to do this but I was struggling with syntax. In either regard, this is still faster than the previous implementation of making a request per ticker

File: sample-portfolio.ini
Comment (ConradSelig) [reply sample-portfolio.ini]: We had it in there just in case someone wanted to start fresh from the base config file. @lemonase originally implemented this feature, but I'm not opposed to removing it as the base file will always be in the repo.

Unless @lemonase has strong reservations about keeping it in I'm fine with the removal.

File: cliStocksTracker.py
Comment (ConradSelig) [reply cliStocksTracker.py:329]: This makes an array ("data") that contains the cost of the stock at each minute where index 0 is market open and index -1 is the most recent minute. The first line gets the column "Open" (the current price at each minute) from the dataframe yfinance returns and the second line strips away the timestamp, leaving only the prices.

If whatever you implemented continues to function than great! Glad to get rid of this rather obscure code.

File: cliStocksTracker.py
Comment (ConradSelig) [reply cliStocksTracker.py]: Still seeing those native round() functions! Looking for utils.round() so the user can choose if they are using the expected round() action or if it should always round down.

**Comment by ConradSelig** [comment]
Overall looks great so far - I spent about 5 minutes initially working with yfinance to get what I needed, never even bothered to look back to see if there was a better way to do it.

Also thought of another performance improvement we could make... see #31, if you want to implement that here awesome! Otherwise don't sweat it, we can catch it later.

Can't wait to deep-dive into the final pull request!

File: sample-portfolio.ini
Comment (lemonase) [reply sample-portfolio.ini]: Yeah I was kinda split between creating a sample config file or generating it in the script. Either way works 4 me.
As long as there's some sort of default sample config to start out with I'm happy.

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:39]: the config is nice but it requires all of the params to be stored as strings. I've created this merger, which admittedly, is a bit wonky but essentially the flow is this
- init all args with defaults
- if the config specifies a section, override the args with that
- only pass args around for source of truth for options

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:65]: created a separate file and class for all screen output for better manageability

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:74]: added defaults for everything - also ran default black linter

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py:200]: moved to utils for better reuse

File: cliStocksTracker.py
Comment (patrick-hovsepian) [inline cliStocksTracker.py]: single download for all ticker data - also wrapped in the try catch from a previous PR

File: renderer.py
Comment (patrick-hovsepian) [inline renderer.py:8]: this class is for all printing - it conforms all rounding to the utils implementation

File: sample-portfolio.ini
Comment (patrick-hovsepian) [reply sample-portfolio.ini]: cool - yeah I'd advocate for this. if/when we make changes to the config layout we can update here rather than the generator function

**Review by patrick-hovsepian — COMMENTED** [review]
highlights

**Comment by ConradSelig** [comment]
@patrick-hovsepian Let me know when you're happy with the pull request, I'm already doing some testing but don't want to accept it before you are done making changes.

**Comment by patrick-hovsepian** [comment]
@ConradSelig should be ready for final approval

File: cliStocksTracker.py
Comment (ConradSelig) [inline cliStocksTracker.py:262]: I would love to see a specific exception here - just so if something else goes wrong the user still gets a stack trace for forward our way.

File: cliStocksTracker.py
Comment (ConradSelig) [inline cliStocksTracker.py:61]: I see at least one black styling commit in here, but this doesn't look styled.

**Review by ConradSelig — CHANGES_REQUESTED** [review]
Other than specified comments looks great! Fix up those two items and I'll make the pull.

File: cliStocksTracker.py
Comment (patrick-hovsepian) [reply cliStocksTracker.py:262]: this was merged from the main branch - i've raised the exception explicitly. if you have something else in mind could you do it as a separate PR?

File: cliStocksTracker.py
Comment (patrick-hovsepian) [reply cliStocksTracker.py:61]: i ran the styler again (commit b224f24) it didn't update this. Do you have a config file you could share or have you gotten different results locally? IMO this shouldn't hold up the PR but i'll defer to you.

File: cliStocksTracker.py
Comment (ConradSelig) [reply cliStocksTracker.py:262]: I'm not going to freak out about it - we'll just fix it later.
This does work! Whats there to complain about?

File: cliStocksTracker.py
Comment (ConradSelig) [reply cliStocksTracker.py:61]: Just doesn't _look_ like black styling... but I'll admit I didn't run the styler on it myself to actually check.
If the styler likes it I like it.

**Pull request merged**
